pennylane_lightning/core/src/simulators/lightning_gpu/StateVectorCudaManaged.hpp

```diff
@@ -359,33 +360,128 @@ class StateVectorCudaManaged
+        if (opName == "MultiRZ") {
+            const std::vector<std::string> names(tgtsInt.size(), {"RZ"});
+            applyParametricPauliGeneralGate_(names, ctrlsInt, ctrls_valuesInt,
+                                             tgtsInt, params.front(), adjoint);
+        } else if (opName == "GlobalPhase") {
+            const std::vector<std::string> names(tgtsInt.size(), "I");
             applyParametricPauliGeneralGate_(names, ctrlsInt, ctrls_valuesInt,
                                              tgtsInt, 2 * params[0], adjoint);
+
+        } else if (native_gates_.find(opName) != native_gates_.end()) {
+            applyParametricPauliGeneralGate_({opName}, ctrlsInt,
+                                             ctrls_valuesInt, tgtsInt,
+                                             params.front(), adjoint);
+        } else if (opName == "Rot") {
+            auto rot_matrix =
+                adjoint
+                    ? cuGates::getRot<CFP_t>(params[2], params[1], params[0])
+                    : cuGates::getRot<CFP_t>(params[0], params[1], params[2]);
+            applyDeviceGeneralGate_(rot_matrix.data(), ctrlsInt, tgtsInt,
+                                    ctrls_valuesInt, adjoint);
+        } else if (par_gates_.find(opName) != par_gates_.end()) {
+            // TODO: offload to par_gates_ if available
+            auto &gateMap =
+                cuGates::DynamicGateDataAccess<PrecisionT>::getInstance();
+            auto &&matrix_cu = gateMap.getGateData(opName, params);
+
+            gate_cache_.add_gate(opName, params[0], matrix_cu);
+
+            applyDeviceGeneralGate_(
+                gate_cache_.get_gate_device_ptr(opName, params[0]), ctrlsInt,
+                tgtsInt, ctrls_valuesInt, adjoint);
+        } else { // No offloadable function call; defer to matrix passing
+            auto &&par =
+                (params.empty()) ? std::vector<Precision>{0.0} : params;
+            if (!gate_cache_.gateExists(opName, par[0]) &&
+                gate_matrix.empty()) {
+                std::string message = "Currently unsupported gate: " + opName +
+                                      " and no matrix provided.";
+                throw LightningException(message);
+            } else if (!gate_cache_.gateExists(opName, par[0])) {
+                std::vector<CFP_t> matrix_cu(gate_matrix.size());
+                std::transform(
+                    gate_matrix.begin(), gate_matrix.end(), matrix_cu.begin(),
+                    [](const std::complex<Precision> &x) {
+                        return cuUtil::complexToCu<std::complex<Precision>>(x);
+                    });
+
+                gate_cache_.add_gate(opName, par[0], matrix_cu);
+            }
+            std::reverse(ctrlsInt.begin(), ctrlsInt.end());
+            std::reverse(tgtsInt.begin(), tgtsInt.end());
+            std::reverse(ctrls_valuesInt.begin(), ctrls_valuesInt.end());
```

Comment:
Could you add code comments about the required wires ordering for the future reference?